packages/labbook/src/components/runtime.tsx

```diff
@@ -262,21 +312,25 @@ export const Runtime: React.FC<IProps> = ({ authoredState, interactiveState, set
           {!readOnly &&
           <div className={css["buttons"]}>
             {
-              (backgroundSource === "upload" || showUploadImageButton) &&
+              ((backgroundSource === "upload" || showUploadImageButton)) &&
               <UploadImage
                 authoredState={authoredState}
-                setInteractiveState={setDrawingStateFn}
+                onUploadImage={handleUploadImage}
                 disabled={disableUI}
                 onUploadStart={onUploadStart}
                 onUploadComplete={onUploadEnd}
+                text={"Upload Image"}
+                useModal={selectedItem?.data?.userBackgroundImageUrl}
+                useModalClick={handleUploadModalClick}
+                showUploadIcon={true}
```

Comment:
This is slightly opinionated and not major, but adding these modal-related props to `UploadImage` component seems a bit strange to me, as this component doesn't really render anything related to the modal. It all happens here.

I think this could be a bit cleaner if this logic was moved to here (or even wrapped in a separate helper component if it becomes too wordy). Basically, we could render `<UploadButton>` with `onClick={handleUploadModalClick}` only if `!!selectedItem?.data?.userBackgroundImageUrl === true`. Otherwise, we'd load the `<UploadImage>` component.

Reply:
Thanks Piotr, yeah I originally had it that way but thought it was a bit too wordy / confusing. But I can change it back.